rinja_derive/src/generator.rs

```diff
@@ -2265,6 +2317,67 @@ impl<'a> Generator<'a> {
     }
 }
 
+fn expr_is_int_lit_plus_minus_one(expr: &WithSpan<'_, Expr<'_>>) -> Option<bool> {
+    fn is_signed_singular<T: Eq + Default, E>(
+        from_str_radix: impl Fn(&str, u32) -> Result<T, E>,
+        value: &str,
+        plus_one: T,
+        minus_one: T,
+    ) -> Option<bool> {
+        Some([plus_one, minus_one].contains(&from_str_radix(value, 10).ok()?))
+    }
+
+    fn is_unsigned_singular<T: Eq + Default, E>(
+        from_str_radix: impl Fn(&str, u32) -> Result<T, E>,
+        value: &str,
+        plus_one: T,
+    ) -> Option<bool> {
+        Some(from_str_radix(value, 10).ok()? == plus_one)
+    }
+
+    let Expr::NumLit(_, Num::Int(value, kind)) = **expr else {
+        return None;
+    };
+    match kind {
+        Some(IntKind::I8) => is_signed_singular(i8::from_str_radix, value, 1, -1),
+        Some(IntKind::I16) => is_signed_singular(i16::from_str_radix, value, 1, -1),
+        Some(IntKind::I32) => is_signed_singular(i32::from_str_radix, value, 1, -1),
+        Some(IntKind::I64) => is_signed_singular(i64::from_str_radix, value, 1, -1),
+        Some(IntKind::I128) => is_signed_singular(i128::from_str_radix, value, 1, -1),
+        Some(IntKind::Isize) => {
+            if cfg!(target_pointer_width = "16") {
```

Comment:
Why not doing `isize::from_str_radix`?

Reply:
The target and host pointer widths may be different.

Reply:
Ah ok, cross-compilation was the issue.